pages/room/[id].tsx

```diff
@@ -1,15 +1,44 @@
-import { EuiText, EuiFlexGroup, EuiFlexItem } from "@elastic/eui";
+import { EuiText, EuiFlexGroup, EuiFlexItem, EuiButton} from "@elastic/eui";
 import { useRouter } from 'next/router';
+import { database } from '../../firebaseConfig';
+import { doc, deleteDoc } from "firebase/firestore";
 
-export default function Room({ }) {
+export default function Room() {
   const router = useRouter()
-  const { id } = router.query
+  const { id, roomid } = router.query
+
+  const deleteRoom = () => {
+    let room = roomid?.toString()
+    deleteDoc(doc(database, "rooms", room))
+      .then(() => {
+        router.push(`/`)
+      })
+  }
```

Comment:
The delete function pulls the FB doc ID out of the query param `roomid`. It sends this roomID to FB to delete the room then forwards the user to the index page.